examples/network_traffic_prediction/nr_models.py

```diff
@@ -0,0 +1,702 @@
+    2: all types of ts features have the same time seria size.
+
+    Return:
+    -------
+    new_st_feats: dict of tensor, or list of dict of tensor
+
+    """
+    if isinstance(input_feats, dict):
+        st_feats = {}
+        for a_type, feats in input_feats.items():
+            # has time seria features
+            if a_type in ts_feat_names:
+                ts_names = ts_feat_names[a_type]
+                num_ts_names = len(ts_names)
+                feats = feats[:, 0: -(num_ts_names * ts_size)]
+            # only has static features, just copy the feature back.
+            else:
+                feats = feats
+            st_feats[a_type] = feats
+    elif isinstance(input_feats, list) and isinstance(input_feats[0], dict):
+        st_feats = []
+        for one_input_feats in input_feats:
+            one_st_feats = {}
+            for a_type, feats in one_input_feats.items():
+                # has time seria features
+                if a_type in ts_feat_names:
+                    ts_names = ts_feat_names[a_type]
+                    num_ts_names = len(ts_names)
+                    feats = feats[:, 0: -(num_ts_names * ts_size)]
+                # only has static features, just copy the feature back.
+                else:
+                    feats = feats            
+                one_st_feats[a_type] = feats
+            st_feats.append(one_st_feats)
+    else:
+        raise NotImplementedError(f'The format of {input_feats} is not a dict or a list dict')
+
+    return st_feats
+
+def get_one_step_ts_feats(input_feats, ts_feat_names, ts_size, window_size, step):
+    """ Extract time seria features only
+
+    edge_feats: list of dict of tuple and tensor
+        A list of edge features for all blocks. Elements are dict in format [tuple, tensor]
+    ts_feat_name: dict of tupe and string
+        A dict of edge type and time serias feature names in format [tuple, string]
+    ts_size: int
+        The overall time seria size, e.g., 24.
+    window_size: int
+        The sliding window size, e.g., 7.
+    step: int
+        The current sliding step.
+    event_name: string
+        Specific argument for the Network Congestion. Default is `event`.
+    
+    Assumption:
+    1: ts features are always be concatinated at the end of edge features; 
+    2: labels only contain time seria values. So need to extract the window + 1 values as labels.
+
+    Return:
+    -------
+    new_ts_edge_feats: list of dict of tuple and tensor
+
+    """
+    if isinstance(input_feats, dict):
+        one_step_ts_feats = {}
+        for a_type, feats in input_feats.items():
+            # has ts features
+            if a_type in ts_feat_names:
+                ts_names = ts_feat_names[a_type]
+                num_ts_names = len(ts_names)
+                ts_feats = feats[:, -(num_ts_names * ts_size): ]
+                # For generasity: if has mutliple ts_feats, mean them into the ts_size
```

Comment:
Yes. This will help to solve the different input dimension issue if two node/edge types have different numbers of time series features. Although there is no such as shown in this air transportation data, it could happen in other dataset we experimented.